### PR TITLE
Auto join fix

### PR DESCRIPTION
**Pull request by parkkoon**
Auto join fix
Delay Occupy
Update Root Balance

### CODE REVIEW

File: app/javascripts/components/playSlot/actions.jsx [hunk 1]
```diff
@@ -147,9 +160,13 @@ export function occupySlotMachine(slotMachineContract, playerAddress, weiValue)
...
-      dispatch({
-        type: ACTION_TYPES.SUCCEEDED_TO_OCCUPY_SLOT_MACHINE,
-      });
+      setTimeout(() => {
+        const ethValue = Web3Service.makeEthFromWei(weiValue);
+        dispatch(updateBalance(ethValue.times(-1)));
+        dispatch({
+          type: ACTION_TYPES.SUCCEEDED_TO_OCCUPY_SLOT_MACHINE,
+        });
+      }, 5000);
```
Comment (fish-sushi) [inline app/javascripts/components/playSlot/actions.jsx]: please use it with constant.
```
const WAIT_TIME_TO_OCCUPY_SLOT_MACHINE
```

File: app/javascripts/components/playSlot/actions.jsx [hunk 2]
```diff
@@ -32,14 +34,25 @@ export const ACTION_TYPES = {
 export function setPlayerKickedByWatcher(slotMachineContractAddress, playerAddress) {
   return async dispatch => {
     try {
-      const transaction = await Web3Service.setPlayerKickedByWatcher(slotMachineContractAddress);
+      const leavedEvent = await Web3Service.setPlayerKickedByWatcher(slotMachineContractAddress);
+      const transaction = Web3Service.web3.eth.getTransaction(leavedEvent.transactionHash);
+      const data = leavedEvent.data.substr(2);
+      const leavedEther = Web3Service.makeEthFromWei(`0x${data.substr(64, 64)}`);
```
Comment (fish-sushi) [inline app/javascripts/components/playSlot/actions.jsx]: The variable name is confusing.
change it more precisely name.

File: app/javascripts/components/slotList/actions.jsx [hunk 3]
```diff
@@ -82,20 +95,23 @@ export function getAllSlotMachines(myAccount) {
...
+              });
+            }
+            slotInformationArr.push(slotInformation);
           }
-
-          slotInformationArr.push(slotInformation);
         }
       }
-
+      dispatch(updateJustLeavedSlotAddress(''));
```
Comment (fish-sushi) [inline app/javascripts/components/slotList/actions.jsx]: I think `null` is better than ''

File: app/javascripts/components/slotList/actions.jsx [hunk 4]
```diff
@@ -82,20 +95,23 @@ export function getAllSlotMachines(myAccount) {
...
-              text: 'You alerady had playing slot Game. Move to that slotMachine.',
-            });
-            dispatch(push(`/slot/play/${slotInformation.meta.address}`));
+        if (slotMachineContractAddress !== justLeavedSlotAddress) {
+          const slotInformation = await getSlotMachineInfo(slotMachineContractAddress, USER_TYPES.PLAYER, myAccount);
+          if (slotInformation && slotInformation.meta) {
+            if (slotInformation.meta.isAlreadyOccupiedByMe) {
+              console.log(slotInformation.meta.slotName);
+              Toast.notie.confirm({
+                text: 'You already had playing slot Game. Do you want to move to that slotMachine?',
```
Comment (fish-sushi) [inline app/javascripts/components/slotList/actions.jsx]: Your last game is still alive. Do you want to move to the last game?

File: app/javascripts/components/slotList/index.jsx [hunk 5]
```diff
@@ -21,22 +21,29 @@ class SlotListContainer extends React.PureComponent {
...
   }
 
   getSlotMachines() {
-    const { dispatch, rootState } = this.props;
-    if (rootState.get('account')) {
-      dispatch(getAllSlotMachines(rootState.get('account')));
+    const { dispatch, rootState, slotListState } = this.props;
+    if (rootState.get('account') && !this.slotListLoaded) {
+      dispatch(getAllSlotMachines(rootState.get('account'), slotListState.get('justLeavedSlotAddress')));
+      if (slotListState.get('justLeavedSlotAddress') !== '') {
```
Comment (fish-sushi) [inline app/javascripts/components/slotList/index.jsx]: you can use `null`

File: app/javascripts/components/slotList/index.jsx [hunk 6]
```diff
@@ -21,22 +21,29 @@ class SlotListContainer extends React.PureComponent {
 
     this.handleToggleDropdown = this.handleToggleDropdown.bind(this);
     this.handleClickSortingOption = this.handleClickSortingOption.bind(this);
+    this.slotListLoaded = false;
```
Comment (fish-sushi) [inline app/javascripts/components/slotList/index.jsx:24]: Why this flag is needed?

File: app/javascripts/components/slotList/reducer.jsx [hunk 7]
```diff
@@ -22,6 +22,7 @@ export const SLOT_LIST_INITIAL_STATE = fromJS({
   mySlotPage: 1,
   isMaking: false,
   setIntervalTimerId: null,
+  justLeavedSlotAddress: '',
```
Comment (fish-sushi) [inline app/javascripts/components/slotList/reducer.jsx]: `null`

**Review by fish-sushi — CHANGES_REQUESTED** [review]

File: app/javascripts/components/playSlot/actions.jsx (hunk 1 above)
Comment (parkkoon) [reply app/javascripts/components/playSlot/actions.jsx]: I agree

File: app/javascripts/components/slotList/index.jsx (hunk 6 above)
Comment (parkkoon) [reply app/javascripts/components/slotList/index.jsx:24]: this is for preventing double loading after initialization about alreadyMachine

**Comment by fish-sushi** [comment]
:sake:

**Pull request merged**